diskpy/ICgen/pos_class.py

```diff
@@ -117,6 +117,67 @@ def _generate_r(self):
         print 'Generating r positions'
         cdf_inv_r = self._parent.sigma.cdf_inv
         
+        if self.method == 'glass':
+            sn = pynbody.load("/home/wadsley/python/glass16.std")
+            #Need to get a cylinder radius 1 and from z=-1 to +1 with self.nParticles in it
+            # Careful density in cylinder not uniform -- increases with radius (fix is to say r=x**2+y**2)
+            #Note: final distribution very flat -- ideally need to know something about h/r ratio
+            #Make a cylinder a little bigger than needed, 
+            #get r of particles, sort and then keep nParticles with smallest r's 
+            #reset last few to be inside if needed
+            #calculate thetas
+            nParticles = self.nParticles
+            hratio = 0.05
+            rscale = 1./(nParticles/(2*np.pi*4096)/hratio)**.3333333333333333
+            zscale = rscale/hratio
+            print "h/R ",hratio," scaling ",rscale,zscale
+            nboxi = int(1./rscale+.6)
+            nboxj = int(1./rscale+.6)
+            nboxk = int(1./zscale+.6)
+            print "nbox",nboxi,nboxj,nboxk
+            npMax = 4096*(nboxi*2+1)*(nboxj*2+1)*(nboxk*2+1)
+            rp = np.zeros(npMax)+1e10
+            thetap = np.zeros(npMax)
+            zp = np.zeros(npMax)
+            nadd = 0
+            for i in range(-nboxi, nboxi+1):
+                for j in range(-nboxj, nboxj+1):
+                   for k in range(-nboxk, nboxk+1):
+                       xloc = (sn.g['x'] + i)*rscale
```

Comment:
If I'm not mistaken, this may produce jagged boundaries at the max radius, since we are approximating a circle in cartesian blocks of 16x16x16.